.github/workflows/go-test.yml

```
@@ -0,0 +1,48 @@
+name: Go tests
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-latest
+          - ubuntu-latest
+        go-version:
+          - 1.19.x
+          - 1.22.x
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Configure git in Windows
+        run: |
+          git config --global core.autocrlf input
+        if: ${{ matrix.os == 'windows-latest' }}
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Static checks with Go vet.
+        run: go vet ./...
+      - name: Static checks with Staticcheck
+        run: go run honnef.co/go/tools/cmd/staticcheck ./...
+      - name: Check imports
+        run: go run golang.org/x/tools/cmd/goimports -w .
+      - name: Test with the Go CLI.
+        run: go test -v ./...
+      - name: Verify repo is up-to-date
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            echo 'Updates required:'
+            git status
+            exit 1
+          fi
```

In other repositories, this is checked without any "if" ([example](https://github.com/elastic/elastic-package/blob/9ea90a15a8f1ae7fd552ee70e8c71a1aa045c410/Makefile#L123C2-L124)) : 

```shell
git update-index --really-refresh
git diff-index --quiet HEAD
```


Should it be added `go mod tidy` here too ? Is it run in some previous step ? To check that there are no pending changes in those files.

```suggestion
          go mod tidy
          if [ -n "$(git status --porcelain)" ]; then
            echo 'Updates required:'
            git status
            exit 1
          fi
```

---

Yeah, good idea using these git commands for the check :+1: Added `go mod tidy` too, thanks!